### PR TITLE
Bump pylint, pydantic, remove "artifacts.yaml" logic

### PR DESCRIPTION
**Pull request by aguschin**
👀 

This will require Studio BE update - we need to remove old logic that works with `artifacts.yaml` - that will make Studio MR backend codebase more easy to work with! To clarify, before we already removed "write" access to old `artifacts.yaml`, but kept "read" access. Now we remove "read" access as well. (Please notice, that will lead to some users losing info about model versions that are 3+ months old, but I hope that's ok).

So once this merged, consider bumping major or minor version, but not the patch.

### CODE REVIEW

**Comment by aguschin** [comment]
cc @amritghimire about changes required in BE ^

**Comment by aguschin** [comment]
@pmrowla, I assume this is what you meant to make it compatible with py-template. There are few things still that should be fixed - like remove top-level silence statements here https://github.com/iterative/gto/blob/bumping-versions/gto/config.py#L1, but overall it should be ready for py-template I hope!

**Comment by aguschin** [comment]
close https://github.com/iterative/gto/issues/400

**Comment by shcheklein** [comment]
Since studio depends for now on MLEM and MLEM depends on the first Pydantic, will those co-exist fine together or does it means we need to upgrade / drop MLEM dependency?

File: .github/workflows/check-test-release.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python: ["3.8", "3.9", "3.10"]
+        python: ["3.9", "3.10", "3.11"]
```
Comment (skshetry) [inline .github/workflows/check-test-release.yml:40]: DVC still supports 3.8 so I don't think we should drop that here too.

**Comment by skshetry** [comment]
We discussed this in today's team meeting and decided not to merge this PR due to the breakages involved. We don't want to risk merging this for now. 

Most of the pylint/mypy changes has been cherry-picked into the main branch, and CI is now running from Python 3.8 - 3.11 versions now. 

I have created #413 that supports both Pydantic v1 and v2, so there's no immediate need to migrate to use pydantic v2 APIs too.

**Pull request closed without merging**